testsuite/model/src/main/java/org/keycloak/testsuite/model/RequireProvider.java

```diff
@@ -41,4 +41,10 @@
      */
     String[] only() default {};
 
+    /**
+     * Specifies provider IDs that does not satisfy this requirement. In other words, there must be another provider
+     * of type {@code value()} for satisfying this requirement.
+     */
+    String[] exclude() default {};
```

Comment:
What happens if both `only` and `exclude` are used?

Reply:
When they are used together both are applied. I updated Javadoc